contrib/linux/bash-completion/dosbox

```diff
@@ -0,0 +1,183 @@
+# bash completion for dosbox                               -*- shell-script -*-
+# shellcheck shell=bash
+# vim: st=2:sw=2:et:ai:ft=bash
+### EDITING NOTES:
+# As this is a bash function, when adding variables or arrays it is important
+# to initialise with local. This prevents the var./array polluting the users
+# env. Please check output of:
+#    declare -x ; declare -a
+# after testing your new completion script
+#
+# Notable exception is COMPREPLY, this needs to persist for communication with
+# bash_completion
+####
+_dosbox()
+{
+  COMPREPLY=()
+  local cur prev words cword
+  _init_completion -s || return
+  local IFS=$'|\t\n'
+  local words_filter="${words[*]:1:$(( cword - 1 ))}"
+
+  ### Main options, note the \ line cont.
+  # using nameref
+  # shellcheck disable=SC2034
+  local -a \
+       _main_opts=(
+                    -h --help -fullscreen -startmapper -noautoexec -securemode
+                    -scaler -forcescaler -lang -machine -socket -exit
+                  )\
+     _repeat_opts=(
+                    -c -conf
+                  )\
+  _exclusive_opts=(
+                    --version --printconf --editconf -resetconf -eraseconf
+                    -opencaptures -erasemapper -resetmapper
+                  )\
+  ####
+
+  # setup nameref to point to array
+  [[ ${prev:0:1} == - ]] && local -n _sub_opts=${prev//+(-)/_}
+
+  ### Arrays holding sub. options
+  # The name should be the trigger option with leading "-"s replaced with "_"
+  # for example, --machine => _machine
+  # latter parts of the script will reference these arrays via a "nameref"
+  # i.e. The last option dictates the available sub. options
+
+  #
+  # shellcheck disable=SC2034
+  local -a \
+         _machine=(
+                    hercules cga tandy pcjr ega vgaonly svga_s3 svga_et3000
+                    svga_et4000 svga_paradise vesa_nolfb vesa_oldvbe
+                  )\
```

Comment:
There are missing options in here. At least cga_mono is missing, maybe more.

Reply:
Good eye in catching that!

Reply:
yes, a few are missing..  also missing from the manpage (which is what I based the script on)

correct list should be
```shell
grep -P "(?<=mtype == \")[^\"]+" src/dosbox.cpp -o
cga
cga_mono
tandy
pcjr
hercules
ega
vga
svga_s3
vesa_nolfb
vesa_oldvbe
svga_et4000
svga_et3000
vga_pvga1a
svga_paradise
vgaonly
```

Reply:
`vga` is commented out, it seemed to be an alias to `svga_s3`. Same for `vga_pvga1a` - it seemed to be an alias for `svga_paradise`.

Thanks for spotting missing value in the man page! I'll prepare a PR to update documentation.

Reply:
Even if `machine = vga` and `machine = vga_pvga1a` are  hidden aliases for `vga` and `svga_paradise` respectively, these are likely to confuse users because users don't learn about machine times from inspecting the source code - they instead use the conf file's list of possible values, which is: `Possible values: hercules, cga, cga_mono, tandy, pcjr, ega, vgaonly, svga_s3, svga_et3000, svga_et4000, svga_paradise, vesa_nolfb, vesa_oldvbe`.

The list of machines in bash-completion is currently:

```
               hercules cga cga_mono tandy pcjr ega vgaonly svga_s3
               svga_et3000 svga_et4000 svga_paradise vesa_nolfb
```

The list is missing `vesa_oldvbe`, so I will add that and resolve this comment.

Reply:
`vesa_oldvbe` added. Resolving.